android/titanium/src/java/org/appcelerator/titanium/util/TiLoadImageManager.java

```diff
@@ -44,106 +39,38 @@ public static TiLoadImageManager getInstance()
 		return _instance;
 	}
 
-	protected TiLoadImageManager()
+	private TiLoadImageManager()
 	{
-		handler = new Handler(this);
-		threadPool = Executors.newFixedThreadPool(THREAD_POOL_SIZE);
+		handler = new Handler(Looper.getMainLooper());
+		threadPool = Executors.newFixedThreadPool(4);
```

Comment:
We could either
- Define number of cores available on the device here
- Use [newCachedThreadPool()](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/Executors.html#newCachedThreadPool) to dynamically scale on workload

Reply:
I changed it to use the number of CPU cores available. I did the same in `TiDownloadManager`.